### PR TITLE
machine/samd21: refactor I2C pin handling to use pins in config struct

### PR DESCRIPTION
**Pull request by aykevl**
Also, pick up pin mode from the pins itself instead of statically configuring it. As an added bonus, this reduces the chances of getting it wrong.

### CODE REVIEW

File: src/machine/board_circuitplay_express.go [hunk 1]
```diff
@@ -94,15 +94,15 @@ const (
 // I2C on the Circuit Playground Express.
 var (
 	// external device
-	I2C0 = I2C{Bus: sam.SERCOM5_I2CM,
-		SDA:     SDA_PIN,
-		SCL:     SCL_PIN,
-		PinMode: PinSERCOM}
```
Comment (aykevl) [inline src/machine/board_circuitplay_express.go]: I believe this should have been `PinSERCOMAlt` (fixed by this PR).

**Comment by deadprogram** [comment]
Looks like this PR is ready to be rebased against `dev` as #597 has been merged.

**Comment by deadprogram** [comment]
Reminder to please rebase this branch. Thanks!

**Comment by aykevl** [comment]
Rebased.

**Comment by deadprogram** [comment]
Tested on Nano33 and worked as expected. Thanks @aykevl now merging.

**Pull request merged**